src/lib/viewers/doc/DocBaseViewer.js

```diff
@@ -1080,9 +1092,13 @@ class DocBaseViewer extends BaseViewer {
                     onAnnotationModeEscape={this.handleAnnotationControlsEscape}
                     onFindBarToggle={this.toggleFindBar}
                     onFullscreenToggle={this.toggleFullscreen}
+                    onPageChange={this.setPage}
+                    onPageSubmit={this.handlePageSubmit}
                     onThumbnailsToggle={this.toggleThumbnails}
                     onZoomIn={this.zoomIn}
                     onZoomOut={this.zoomOut}
+                    pageCount={this.pdfViewer.pagesCount}
+                    pageNumber={this.pdfViewer.currentPageNumber}
```

Comment:
When does `renderUI` get called? Should there be some logic to handle the existing pageControls instance in this method?

Reply:
Added!